components/brave_rewards/resources/ui/components/pageWallet.tsx

```diff
@@ -67,6 +69,43 @@ class PageWallet extends React.Component<Props, State> {
     })
   }
 
+  onModalBackupOnCopy = (backupKey: string) => {
+    const success = clipboardCopy(backupKey)
+    // TODO(jsadler) possibly flash a message that copy was completed
+    console.log(success ? 'Copy successful' : 'Copy failed')
+  }
+
+  onModalBackupOnPrint = (backupKey: string) => {
+    let printWindow = window.open(backupKey)
```

Comment:
why is `backupKey` set as the URL of the window to open?

Reply:
actually could you just replace this whole thing with:

```
const printWindow = window.open(`data:,${encodeURIComponent(this.constructBackup(backupKey))}`
printWindow.print()
printWindow.close()
```

Reply:
👍 Exactly what I was looking for. Thanks!

Reply:
or if you can't open data URLs for some reason, replace the data URL with an object URL created like in https://github.com/brave/brave-core/pull/816/files#diff-aa81b445f3d957d5c028d1df07d3ac64R102

Reply:
Will do. (Yes, I get error that I cannot navigate top frame to data URL)

Reply:
We can't open data URLs directly anymore: https://groups.google.com/a/chromium.org/forum/#!topic/blink-dev/GbVcuwg_QjM%5B1-25%5D

and when I replace it with the createURLObject it offers to download it when clicking the print button.

The alternatives in the link suggest creating a iframe for the data but I think that puts us back in XSS land

Reply:
@jasonrsadler i found a workaround. basically you can avoid `document.write` if the opened window is same-origin as the parent because then you can access DOM methods on the child window from the opener. so it becomes:

```
const win = window.open(document.location)
win.document.body.innerText = this.constructBackup(backupKey) // this should be text, not HTML
win.print()
```

Reply:
This works good with some modifications:
```
if (document.location) {
      const win = window.open(document.location.href)
      if (win) {
        win.document.body.innerText = utils.constructBackupString(backupKey) // this should be text, not HTML
        win.print()
        win.close()
      }
    }
```
However it does cause an error in the console. What would be a safe and still working url to pass through?
```
"%cUnhandled rejection Error: No tab url specified
    at t.getShieldSettingsForTabData (chrome-extension://mnojpmjdmbbfmejpflffifhffcmidifd/js/background.bundle.js:1:30562)
    at l (chrome-extension://mnojpmjdmbbfmejpflffifhffcmidifd/js/background.bundle.js:1:110001)
    at I._settlePromiseFromHandler (chrome-extension://mnojpmjdmbbfmejpflffifhffcmidifd/js/background.bundle.js:1:83232)
    at I._settlePromise (chrome-extension://mnojpmjdmbbfmejpflffifhffcmidifd/js/background.bundle.js:1:84032)
    at I._settlePromise0 (chrome-extension://mnojpmjdmbbfmejpflffifhffcmidifd/js/background.bundle.js:1:84731)
    at I._settlePromises (chrome-extension://mnojpmjdmbbfmejpflffifhffcmidifd/js/background.bundle.js:1:86083)
    at chrome-extension://mnojpmjdmbbfmejpflffifhffcmidifd/js/background.bundle.js:1:39533
From previous event:
    at I.B [as _captureStackTrace] (chrome-extension://mnojpmjdmbbfmejpflffifhffcmidifd/js/background.bundle.js:1:53021)
    at I._then (chrome-extension://mnojpmjdmbbfmejpflffifhffcmidifd/js/background.bundle.js:1:78689)
```

Reply:
@jasonrsadler that seems like maybe a separate bug? if it's not causing any problems i think we can just ignore it for now

Reply:
I was thinking that but I hadn't seen it before. I'll leave it alone for now and revisit if it's something my code can fix.

Reply:
Done.

Reply:
i think the issue there is that https://github.com/brave/brave-extension/blob/0a628254dd6f7d5990ddf145626587c083c82fc0/app/background/api/shieldsAPI.ts#L83 should catch the error and handle it